ext/sockets/sockets.c

```diff
@@ -2589,61 +2591,67 @@ PHP_FUNCTION(socket_addrinfo_lookup)
 #  endif
 #endif
 
-	if (zhints && !HT_IS_PACKED(Z_ARRVAL_P(zhints))) {
-		ZEND_HASH_MAP_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(zhints), key, hint) {
-			if (key) {
-				bool failed = false;
-				if (zend_string_equals_literal(key, "ai_flags")) {
-					zend_long val = zval_try_get_long(hint, &failed);
-					if (failed) {
-						zend_argument_type_error(3, "\"ai_flags\" key must be of type int, %s given", zend_zval_type_name(hint));
+	if (zhints) {
+		if (!HT_IS_PACKED(Z_ARRVAL_P(zhints))) {
```

Comment:
Maybe flip the condition and don't have an else branch?
```suggestion
		if (UNEXPECTED(HT_IS_PACKED(Z_ARRVAL_P(zhints)))) {
		/* Throw error */
		}
		/* Happy path */
```